internal/dataplane/kongstate/plugin.go

```diff
@@ -18,33 +18,34 @@ import (
 )
 
 // getKongPluginOrKongClusterPlugin fetches a KongPlugin or KongClusterPlugin (as fallback) from the store.
-// If both are not found, an error is returned.
+// If both don't exist pluginFound is set to false.
```

Comment:
WDYT about not adding the boolean return value but instead return `store.NotFoundError` when neither plugin nor cluster plugin are found and then handle that error type at the call site checking the type of the error?

Reply:
I followed convention with maps that you have a boolean to indicate whether a value exists or not. Now it makes an error reserved only for something that feels to me like a "real" error and does not require from a caller understanding of dedicated error for checking existence. 

I always feel a little bit uncomfortable when errors are used to report something that may be a result of normal usage. In a language with proper discriminated unions (F# naming), variants/tagged unions (OCaml naming), or enums in Rust, I can model it with one return value that can be `KP | KCP | None | Err` and deconstruct it with exhaustive pattern matching. Unfortunately, Go does not offer it, so for me the most sensible/readable was to have a dedicated boolean to indicate whether something was found or not without errors

So to sum up, IMHO it can be either way in Go :D

Reply:
Fair enough 👍 Let's leave it as is.